engine/runtime/src/main/scala/org/enso/compiler/phase/SymbolsImportResolution.scala

```
@@ -0,0 +1,562 @@
+  /**
+   * From the given module bindings, finds all the static module method and type definitions.
+   */
+  private def findAllStaticMethodsAndTypesDefinedInModule(
+    moduleBindings: List[IR.Module.Scope.Definition]
+  ): List[IR.Name.Literal] = {
+    logger.log(
+      Level.FINER,
+      "Finding all static methods and types in module"
+    )
+    val symbolsToImport: ListBuffer[IR.Name.Literal] = ListBuffer.empty
+    moduleBindings.foreach {
+      case Definition.Method.Explicit(methodRef, _, _, _, _)
+        if methodRef.methodName.isInstanceOf[Name.Literal] =>
+        symbolsToImport += methodRef.methodName.asInstanceOf[Name.Literal]
+      case Definition.Method.Binding(methodRef, _, _, _, _, _)
+        if methodRef.methodName.isInstanceOf[Name.Literal] =>
+        symbolsToImport += methodRef.methodName.asInstanceOf[Name.Literal]
+      case Definition.Type(name, _, _, _, _, _)
+        if name.isInstanceOf[Name.Literal] =>
+        symbolsToImport += name.asInstanceOf[Name.Literal]
+      case _ => ()
+    }
+    symbolsToImport.toList
+  }
+
+  private def findAllExportedSymbolsFromModule(
+    importedModuleBindingMap: BindingsMap
+  ): List[BindingsMap.ImportTarget] = {
+    val exportedSymbols: ListBuffer[BindingsMap.ImportTarget] = ListBuffer.empty
+    importedModuleBindingMap.exportedSymbols.foreach {
+      case (_, resolvedNames) =>
+        resolvedNames.foreach {
+          case resolvedType: BindingsMap.ResolvedType => exportedSymbols += resolvedType
+          case resolvedModule: BindingsMap.ResolvedModule => exportedSymbols += resolvedModule
+          case resolvedMethod: BindingsMap.ResolvedMethod => exportedSymbols += resolvedMethod
+          case resolvedCtor: BindingsMap.ResolvedConstructor => exportedSymbols += resolvedCtor
+        }
+    }
+    exportedSymbols.toList
+  }
+
+  /**
+   * Finds all teh constructors from a type. Returns empty list if there is no such type,
+   * or if the type does not have any constructors.
```

```suggestion
   * Finds all constructors from a type. Returns empty list if there is no such type,
   * or if the type does not have any constructors.
```

Also, should probably return an error if type does not exist

---

As noted in the comment - the error will be reported later. It is OK for this method to return an empty list in case the type does not exist.